R/bmerge.R

```diff
@@ -34,86 +34,120 @@ bmerge = function(i, x, icols, xcols, roll, rollends, nomatch, mult, ops, verbos
     ans
   }
 
+  cast_with_atts = function(x, as.f) {
```

Comment:
should these helpers be declared in the {data.table} namespace instead (not exported)?

if `bmerge()` gets called dozens/hundreds of times, these helpers would have to be re-created each time. my approach is usually to put such helpers outside function bodies unless it's strongly beneficial for them to inherit directly from the function environment (not the case here AFAICT).

Reply:
oh, I see `coerce_col` inherits `verbose`.

Suggestion: `verbose_msg = NULL`, then `if (!is.null(verbose_msg))` to replace `if (verbose)`

Reply:
done